AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/ModelSyncedPayload.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+struct ModelSyncedPayload {
+    var modelName: String
+    var isFullSync: Bool
+    var isDeltaSync: Bool
```

Comment:
Do the other platforms have two properties to represent this? These are mutually exclusive states; we should either model it as a single bool, or an enum with two cases

Reply:
https://github.com/aws-amplify/amplify-js/blob/eb9719467f7c143a26bfd2d598b3903ff0815731/packages/datastore/src/sync/index.ts#L569
https://github.com/aws-amplify/amplify-android/pull/710/files#diff-bad527d7894aa87e0315e7104f460145R53

Reply:
As discussed, we will leave these as two properties on the public payload, although we'll represent them in the builder with an enum